### ISSUE TITLE
training of the modified vgg11 model

### ISSUE THREAD

**Issue by johngrabner** [cwig/start_follow_read#5]
Firstly, thank you for your paper "Start, Follow, Read: Full-Page End-to-end Handwriting Recognition".  

I am trying to get a better understanding of the training of the modified vgg11 model you used in your paper.  To this end, I translated your vg11 found in vgg.py does into Keras. 

   ```
inputShape = (height, width, depth)
   inputs = Input(shape=inputShape)
   # Conv Block 1
   x = Conv2D(64, (3, 3), activation='relu', padding='same')(inputs)
   x = MaxPooling2D(pool_size=(2, 2), strides=(2, 2))(x)

   # Conv Block 2
   x = Conv2D(128, (3, 3), activation='relu', padding='same')(x)
   x = MaxPooling2D(pool_size=(2, 2), strides=(2, 2))(x)

   # Conv Block 3
   x = Conv2D(256, (3, 3), activation='relu', padding='same')(x)
   x = Conv2D(256, (3, 3), activation='relu', padding='same')(x)
   x = MaxPooling2D(pool_size=(2, 2), strides=(2, 2))(x)

   # Conv Block 4
   x = Conv2D(512, (3, 3), activation='relu', padding='same')(x)
   x = Conv2D(512, (3, 3), activation='relu', padding='same')(x)
   x = MaxPooling2D(pool_size=(2, 2), strides=(2, 2))(x)

   # Conv Block 5
   x = Conv2D(512, (3, 3), activation='relu', padding='same')(x)
   x = Conv2D(512, (3, 3), activation='relu', padding='same')(x)

   # FC layers
   x = Flatten()(x)
   x = Dense(4, activation='linear')(x)

   model = Model(inputs, x)
```

I trained this model with 5000 test images of a handwritten script and corresponding bounding box for a word off center as a baby step to replicate your SOL target approach.  
```
sgd = keras.optimizers.SGD(lr=0.001, decay=1e-6, momentum=0.9, nesterov=True)
model.compile(loss=losses.mean_squared_error, optimizer=sgd)
```

To keep things simple to start, my training data only present a subset of  SOL data plus width and height.  This modified vgg11 only learns the average the SOL data rather than a model that can predict sol parameters based on an input image.

Here are some samples of my output with green being ground thruth and red being predicted value from the vgg11 model:
```
predictY = model.predict(testX)
print_sample(testX, testY, predictY)
```
![image](https://user-images.githubusercontent.com/8209285/58202804-24778380-7c9e-11e9-88e4-b4f40107d6fe.png)

![image](https://user-images.githubusercontent.com/8209285/58202816-2c372800-7c9e-11e9-91ad-e65377a815fa.png)

![image](https://user-images.githubusercontent.com/8209285/58202831-33f6cc80-7c9e-11e9-8ae3-285880cd8fe9.png)

I would appreciate some suggestions as to why my copy vgg11 can't seem to learn anything about the potential SOL words presented to it or some insight into how your modified vgg11 was trained.

Regards,
John Grabner

**Comment by johngrabner** [comment]
It would appear that training vgg11 from scratch using regression makes it very sensitive to the distribution of the variables being regressed.  A little normalization of the training data and things got a lot better.

![image](https://user-images.githubusercontent.com/8209285/58217205-518c5c00-7cc7-11e9-86cd-d5b46edec657.png)

![image](https://user-images.githubusercontent.com/8209285/58217222-679a1c80-7cc7-11e9-9578-bb8658f67181.png)

![image](https://user-images.githubusercontent.com/8209285/58217235-7ed90a00-7cc7-11e9-86b8-7af97cca618c.png)

Additional work on improving the distribution of my training data is obviously needed. In any event, I would appreciate if you could share more details on how you trained the vgg11
